packages/account-v2/src/modules/FinancialAssessment/FinancialAssessmentForm.tsx

```diff
@@ -0,0 +1,228 @@
+    const { data: accountStatus } = useAccountStatus();
+    const { isEUCountry } = useIsEuRegion();
+    const { isMobile } = useDevice();
+    const history = useHistory();
+
+    const [isConfirmModalOpen, setIsConfirmModalOpen] = useState(false);
+
+    const redirect = isNavigationFromDerivGO() || isNavigationFromP2P();
+    const redirectPlatformName = isNavigationFromDerivGO() ? 'Deriv GO' : 'Deriv P2P';
+    const redirectUrl = isNavigationFromDerivGO() ? DERIV_GO_URL : P2P_URL;
+
+    const {
+        is_authenticated: isAuthenticated,
+        is_financial_information_not_complete: isFinancialInformationNotComplete,
+        is_trading_experience_not_complete: isTradingExperienceNotComplete,
+    } = accountStatus || {};
+    const hasTradingExperience = isEUCountry || isTradingExperienceNotComplete;
+
+    const {
+        data: financialAssessment,
+        error: fetchError,
+        isLoading: isFinancialAssessmentLoading,
+        mutation: {
+            error: financialAssessmentUpdateError,
+            isLoading: isFinancialAssessmentUpdating,
+            isSuccess: isFinancialAssessmentUpdateSuccess,
+        },
+        update: updateFinancialAssessment,
+    } = useFinancialAssessment();
+
+    const {
+        account_turnover: accountTurnover,
+        binary_options_trading_experience: binaryOptionsTradingExperience,
+        binary_options_trading_frequency: binaryOptionsTradingFrequency,
+        cfd_trading_experience: cfdTradingExperience,
+        cfd_trading_frequency: cfdTradingFrequency,
+        education_level: educationLevel,
+        employment_industry: employmentIndustry,
+        employment_status: employmentStatus,
+        estimated_worth: estimatedWorth,
+        forex_trading_experience: forexTradingExperience,
+        forex_trading_frequency: forexTradingFrequency,
+        income_source: incomeSource,
+        net_income: netIncome,
+        occupation,
+        other_instruments_trading_experience: otherInstrumentsTradingExperience,
+        other_instruments_trading_frequency: otherInstrumentsTradingFrequency,
+        source_of_wealth: sourceOfWealth,
+    } = financialAssessment;
+
+    const initialValues = {
+        accountTurnover,
+        educationLevel,
+        employmentIndustry,
+        employmentStatus,
+        estimatedWorth,
+        incomeSource,
+        netIncome,
+        occupation,
+        sourceOfWealth,
+        ...(hasTradingExperience && {
+            binaryOptionsTradingExperience,
+            binaryOptionsTradingFrequency,
+            cfdTradingExperience,
+            cfdTradingFrequency,
+            forexTradingExperience,
+            forexTradingFrequency,
+            otherInstrumentsTradingExperience,
+            otherInstrumentsTradingFrequency,
+        }),
+    };
+
+    type TFinancialAssessmentFormValues = typeof initialValues;
+
+    const onSubmit = (handleSubmit: (e?: React.FormEvent<HTMLFormElement> | undefined) => void) => () => {
```

Comment:
do we need to specify  `undefined` since the parameter is optional

Reply:
updated.